modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/add_to_order/page.jsp

```diff
@@ -18,32 +18,45 @@
 
 <div class="add-to-cart mb-2" id="<%= addToCartId %>">
 	<liferay-util:include page="/add_to_order/skeleton.jsp" servletContext="<%= application %>" />
-</div>
 
-<aui:script require="commerce-frontend-js/components/add_to_cart/entry as AddToCart">
-	const initialProps = {
-		channel: {
-			currencyCode: '<%= currencyCode %>',
-			id: <%= channelId %>,
-		},
-		cpInstance: {
-			accountId: <%= commerceAccountId %>,
-			inCart: <%= inCart %>,
-			options: '<%= options %>',
-			skuId: <%= skuId %>,
-			stockQuantity: <%= stockQuantity %>,
-		},
-		orderId: <%= orderId %>,
-		settings: {
-			block: <%= block %>,
-			disabled: <%= disabled %>,
-			willUpdate: <%= willUpdate %>,
-			withQuantity: {
-				forceDropdown: true,
-			},
-		},
-		spritemap: '<%= spritemap %>',
-	};
-
-	AddToCart.default('<%= addToCartId %>', '<%= addToCartId %>', initialProps);
-</aui:script>
\ No newline at end of file
+	<react:component
+		module="add_to_order/AddToOrder"
+		props='<%=
+			HashMapBuilder.<String, Object>put(
+				"channel", HashMapBuilder.<String, Object>put(
+					"currencyCode", currencyCode
+				).put(
+					"id", channelId
+				).build()
+			).put(
+				"cpInstance", HashMapBuilder.<String, Object>put(
+					"accountId", commerceAccountId
+				).put(
+					"inCart", inCart
+				).put(
+					"options", options
+				).put(
+					"skuId", skuId
+				).put(
+					"stockQuantity", stockQuantity
+				).build()
+			).put(
+				"orderId", orderId
+			).put(
+				"settings", HashMapBuilder.<String, Object>put(
+					"block", block
+				).put(
+					"disabled", disabled
+				).put(
+					"willUpdate", willUpdate
+				).put(
+					"withQuantity", HashMapBuilder.<String, Object>put(
+						"forceDropdown", true
+					).build()
+				).build()
+			).put(
+				"spritemap", spritemap
```

Comment:
I left this one in just in case, but you shouldn't need to pass in `spritemap` since it's injected using the `IconSpritemapContext` when we render.